### PR TITLE
feat: Add support for Postgresql dialect

### PR DESCRIPTION
**Pull request by ansh0l**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-spanner/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/spanner_admin_database_v1/types/backup.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ class State(proto.Enum):
         number=12,
         message=timestamp_pb2.Timestamp,
     )
+    database_dialect = proto.Field(proto.ENUM, number=10, enum=common.DatabaseDialect,)
```
Comment (IlyaFaer) [inline google/cloud/spanner_admin_database_v1/types/backup.py:193]: Using just `10` can be a bit confusing for new users. Probably should be done with a constant with a describing name.

File: google/cloud/spanner_v1/instance.py [hunk 2]
```diff
@@ -428,6 +429,7 @@ def database(
         pool=None,
         logger=None,
         encryption_config=None,
+        database_dialect=DatabaseDialect.DATABASE_DIALECT_UNSPECIFIED
```
Comment (IlyaFaer) [inline google/cloud/spanner_v1/instance.py:432]: Docstrings for the new argument should be added

File: tests/system/test_dbapi.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 from google.cloud import spanner_v1
 from google.cloud._helpers import UTC
 from google.cloud.spanner_dbapi.connection import connect
+from google.cloud.spanner_admin_database_v1 import DatabaseDialect
```
Comment (IlyaFaer) [inline tests/system/test_dbapi.py:24]: Seems like the imported class is not used anywhere

File: google/cloud/spanner_admin_database_v1/types/backup.py (hunk 1 above)
Comment (IlyaFaer) [reply google/cloud/spanner_admin_database_v1/types/backup.py:193]: And I don't think `black` formatter will allow such a comma in the end

Comment (asthamohta) [reply google/cloud/spanner_admin_database_v1/types/backup.py:193]: Aren't these auto generated changes? Should the PR include these changes?

File: google/cloud/spanner_v1/backup.py [hunk 4]
```diff
@@ -216,6 +217,13 @@ def referencing_backups(self):
         """
         return self._referencing_backups
 
+    def database_dialect(self):
+        """Encryption info for this backup.
+        :rtype: :class:`~google.cloud.spanner_admin_database_v1.types.DatabaseDialect`
+        :returns: a class representing the dialect of this backup's database
+        """
+        return self._encryption_info
```
Comment (asthamohta) [inline google/cloud/spanner_v1/backup.py:225]: returns the encryption, please change to dialect

File: google/cloud/spanner_v1/types/type.py [hunk 5]
```diff
@@ -59,6 +59,7 @@ class TypeAnnotationCode(proto.Enum):
     the way value is serialized.
     """
     TYPE_ANNOTATION_CODE_UNSPECIFIED = 0
+    INT32 = 1
```
Comment (asthamohta) [inline google/cloud/spanner_v1/types/type.py:62]: This is an unsupported type, should not be added here

File: tests/system/conftest.py [hunk 6]
```diff
@@ -59,7 +72,9 @@ def spanner_client():
             credentials=credentials,
         )
     else:
-        return spanner_v1.Client()  # use google.auth.default credentials
+        return spanner_v1.Client(
+            client_options={"api_endpoint": "staging-wrenchworks.sandbox.googleapis.com"}
```
Comment (asthamohta) [inline tests/system/conftest.py:76]: Please remove staging endpoint

File: tests/system/test_dbapi.py [hunk 7]
```diff
@@ -39,9 +40,8 @@
     PRIMARY KEY (contact_id)""",
 )
 
-
```
Comment (asthamohta) [inline tests/system/test_dbapi.py:42]: nit: looks like the line was accidentally removed

File: tests/_fixtures.py [hunk 8]
```diff
@@ -120,7 +120,54 @@
...
+    name VARCHAR(1024),
+    value BIGINT,
+    PRIMARY KEY (name));
+CREATE TABLE string_plus_array_of_string (
+    id BIGINT,
+    name VARCHAR(16),
+    PRIMARY KEY (id));
+CREATE INDEX name ON contacts(first_name, last_name);
+"""
+# CREATE TABLE contact_phones (
```
Comment (asthamohta) [inline tests/_fixtures.py:150]: Why is this code commented?

File: tests/system/test_backup_api.py [hunk 9]
```diff
@@ -93,11 +93,11 @@ def database_version_time(shared_database):
 
 
 @pytest.fixture(scope="session")
-def second_database(shared_instance, database_operation_timeout):
+def second_database(shared_instance, database_operation_timeout, database_dialect):
     database_name = _helpers.unique_id("test_database2")
     pool = spanner_v1.BurstyPool(labels={"testcase": "database_api"})
     database = shared_instance.database(
-        database_name, ddl_statements=_helpers.DDL_STATEMENTS, pool=pool
+        database_name, ddl_statements=_helpers.DDL_STATEMENTS, pool=pool, database_dialect=database_dialect
```
Comment (asthamohta) [inline tests/system/test_backup_api.py:100]: For PG, first you create a database and then apply update ddl. It doesn't allow passing ddl statements at creation time. This code should most likely fail

**Comment by asthamohta** [comment]
image
From the number of tests run, it looks like PG tests did not run at all

File: google/cloud/spanner_v1/instance.py (hunk 2 above)
Comment (ansh0l) [reply google/cloud/spanner_v1/instance.py:432]: Added

**Comment by ansh0l** [comment]
Regarding `From the number of tests run, it looks like PG tests did not run at all` -> Additional tests are being run, confirmed in meet session.

File: google/cloud/spanner_v1/types/transaction.py [hunk 10]
```diff
@@ -30,7 +30,7 @@
```
Comment (asthamohta) [inline google/cloud/spanner_v1/types/transaction.py:31]: This file is unrelated, can you checkout this file?

**Review by asthamohta — APPROVED** [review]

**Comment by ansh0l** [comment]
adding automerge since kokoro build is successful. But others are stuck.
